gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/membership/domain_service/PrimaryOwnerDomainService.java

```diff
@@ -64,12 +67,19 @@ public void createIntegrationPrimaryOwnerMembership(String integrationId, Primar
             });
     }
 
-    public Maybe<PrimaryOwnerEntity> getApiPrimaryOwner(final String organizationId, String integrationId)
-        throws ApiPrimaryOwnerNotFoundException {
+    public Maybe<PrimaryOwnerEntity> getApplicationPrimaryOwner(final String organizationId, String applicationId) {
```

Comment:
What about using the name `findApplicationPrimaryOwner`? We have this implicit rule where a `find` method return an optional where a `get` method would throw if the resource to find does not exists 🤔